### PR TITLE
=str #15859 Eliminate usage of Stash mailbox

### PR DESCRIPTION
**Pull request by patriknw**
- As reported in #16522 Stash is slow
- Nothing fancy here, but should be more efficient than Stash

### CODE REVIEW

**Comment by drewhk** [comment]
Thanks, this looks like a very good first step!

LGTM

**Comment by patriknw** [comment]
yes, that was my thought, low hanging improvement with a lot of value.
I will not close the ticket, since this doesn't really solve what is described there.

**Comment by drewhk** [comment]
Yep, great work!

**Comment by drewhk** [comment]
Double checked again, and it still looks great :)

File: akka-stream/src/main/scala/akka/stream/impl/ExposedPublisherReceive.scala [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+private[akka] abstract class ExposedPublisherReceive(activeReceive: Actor.Receive, unhandled: Any ⇒ Unit) extends Actor.Receive {
+  private var stash = List.empty[Any]
+
+  def isDefinedAt(o: Any): Boolean = true
+
+  def apply(o: Any): Unit = o match {
+    case ep: ExposedPublisher ⇒
+      receiveExposedPublisher(ep)
+      // we don't use sender() so this is allright
+      stash.reverse.foreach { msg ⇒
```
Comment (patriknw) [inline akka-stream/src/main/scala/akka/stream/impl/ExposedPublisherReceive.scala]: I will add a `if (stash.nonEmpty)` here, just to make sure that there is no closure instance created for the normal case

**Comment by patriknw** [comment]
See #15859

**Pull request merged**